orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -215,7 +239,10 @@ def _create_training_session(self):
         if self._device.type == 'cuda':
             # Configure the InferenceSessions to use the specific GPU on which the model is placed.
             providers = ["CUDAExecutionProvider", "CPUExecutionProvider"]
-            provider_options = [{"device_id": str(self._device.index)}, {}]
+            if self._use_external_cuda_allocator:
+                provider_options = [{"device_id": str(self._device.index), "cuda_external_alloc": str(self._torch_alloc), "cuda_external_free": str(self._torch_free)}, {}]
```

Comment:
```suggestion
                provider_options = [{"device_id": str(_utils.get_device_index(self._device)), "cuda_external_alloc": str(self._torch_alloc), "cuda_external_free": str(self._torch_free)}, {}]
```

Reply:
Thanks, @thiagocrepaldi , seems like a good suggestion but the standard software engineering practice would be to make this change as a different PR since I'm not touching this particular piece of code (i.e `"device_id": str(self._device.index)`). Your suggestion is a clean up and I do not want to mix it with enabling an external allocator.